### PR TITLE
app.json: add env vars

### PR DESCRIPTION
**Pull request by rlewis-grax**
Adds the environment variables that can be automatically configured.

### CODE REVIEW

**Comment by rlewis-grax** [comment]
Tested:

Screenshot 2024-05-16 at 5 58 49 PM
Screenshot 2024-05-16 at 6 01 40 PM

File: app.json [hunk 1]
```diff
@@ -14,5 +14,19 @@
       "plan": "heroku-postgresql",
       "as": "DATABASE"
     }
-  ]
+  ],
+  "env": {
+    "SECRET_STORE_BASE": {
+      "required": true,
+      "generator": "secret"
```
Comment (yonderblue) [inline app.json:21]: pretty cool if its sticky

Comment (yonderblue) [reply app.json:21]: Perhaps take these couple out of the guide then if done automatically. Other stuff in here isn't mentioned either in the same vein.

**Review by yonderblue — APPROVED** [review]

File: app.json (hunk 1 above)
Comment (rlewis-grax) [reply app.json:21]: The guide and the app.json unfortunately cover separate processes. This config is used if you use the "quick-deploy" buttons but if you want to create an app from the group up with the CLI, it doesn't apply this configuration from the repo.

Comment (rlewis-grax) [reply app.json:21]: I stand corrected; we could re-use the app.json to create apps with the CLI, but it’s via a Heroku CLI addon:
```
heroku plugins:install api
heroku api POST /apps --body '{"name": "example-app"}'
```

Comment (yonderblue) [reply app.json:21]: not sure why quick deploy wouldn't suffice for all?

**Pull request merged**